src/actions/AuthActions.js

```diff
@@ -15,23 +18,82 @@ import {tokens, navKeys} from '../constants';
 
 import {persistor} from '../store';
 
-const userLoginSuccess = (dispatch, email, navigate) => {
+const userLoginSuccess = ({dispatch, user, navigate}) => {
     dispatch({
         type: FACEBOOK_LOGIN_SUCCESS,
-        payload: {token: email, appReady: false}
+        payload: {token: user.email, currentUser: user, appReady: false}
     });
 
     navigate(navKeys.SEARCH);
 };
 
-const userLoginFailure = (dispatch, navigate) => {
+const userLoginFailure = ({dispatch, navigate}) => {
     dispatch({
         type: FACEBOOK_LOGIN_FAILURE
     });
 
     navigate(navKeys.LOGIN);
 };
 
+const getUserFavorites = ({dispatch, currentUser, navigate}) => {
+    const {name, email, imageUrl} = currentUser;
+
+    firebase.firestore().collection(`users/${email}/favorites`)
+        .get()
+        .then(querySnapshot => {
+            const userFavorites = _.map(querySnapshot.docs, doc => {
+                let preparedFavorite = doc.data();
+                preparedFavorite.favoriteIsComplete = false;
+
+                return preparedFavorite;
+            });
+
+            currentUser.favorites = userFavorites;
+        })
+        .then(() => {
+            userLoginSuccess({dispatch, user: currentUser, navigate});
+        })
+        .catch(err =>{
+            console.log(err);
+        });
+};
+
+const createUserInFirestore = ({dispatch, user, navigate}) => {
+    const {name, email, imageUrl} = user;
+    const currentUser = {
+        name,
+        email,
+        imageUrl
+    };
+
+    firebase.firestore().doc(`users/${email}`)
+        .set(currentUser)
+        .then(() => {
+
+            getUserFavorites({dispatch, currentUser, navigate});
+        })
+        .catch(err => {
+            console.log(err);
+            userLoginFailure({dispatch, navigate});
+        })
+};
+
+const getFirestoreUserObject = ({dispatch, user, navigate}) => {
+    const {name, email, imageUrl} = user;
+
+    firebase.firestore().collection('users').doc(`${email}`)
+        .get()
+        .then(doc => {
+            if (doc.exists) {
+                const currentUser = doc.data();
+
+                getUserFavorites({dispatch, currentUser, navigate});
+            } else {
+                createUserInFirestore({dispatch, user, navigate});
+            }
```

Comment:
Once a user successfully logs in with facebook we end up here. If they exist in firebase, we grab their favorites. If they don't, then we create them in firebase, and then grab their favorites. If they have no favorites, it'll come back as an empty array.